Tensile/kernelConverter.py

```diff
@@ -0,0 +1,317 @@
+          
+          # remove the "" from the string
+          DefaultValue = re.sub("[\"\']", "", DefaultValue)
+          if str(forkValue) != DefaultValue:
+            YamlFileData.append("         - {}: {}\n".format(forkKey, [forkValue]))
+  
+def Form9BitMIInstruction(currentSolutionState):
+
+   MIBlock = currentSolutionState["MIBlock"]
+   #print1(" MIBlock : {0} ".format(MIBlock))
+  
+   MIWaveTile =  currentSolutionState["MIWaveTile"]
+   MIWaveGroup = currentSolutionState["MIWaveGroup"]
+   #print1(" MIWaveTile : {0} ".format(MIWaveTile))
+   #print1(" MIWaveGroup : {0} ".format(MIWaveGroup))
+    
+   if len(MIBlock) == 0 or len(MIWaveTile) == 0  or len(MIWaveGroup) == 0:
+    raise RuntimeError("Length of MIBlock:{0}, MIWave Tile: {1},MIWaveGroup : {2} cannot be empty".format(len(MIBlock),len(MIWaveTile),len(MIWaveGroup)))
+    
+   MIBlock1 = [MIBlock[i] for i in (0,1,2,3,4)]
+   MIBlock5bits = ','.join([str(item) for item in MIBlock1])
+   #print1(" MIBlock5bits : {0} ".format(MIBlock5bits))
+    
+   MIWaveTile2Bits = ','.join([str(item) for item in MIWaveTile])
+   #print1(" MIWaveTile2Bits : {0} ".format(MIWaveTile2Bits))
+    
+   MIWaveGroup2Bits = ','.join([str(item) for item in MIWaveGroup])
+   #print1(" MIWaveGroup2Bits : {0} ".format(MIWaveGroup2Bits))
+   
+   if len(MIBlock5bits) == 0 or len(MIWaveTile2Bits) == 0  or len(MIWaveGroup2Bits) == 0:
+       raise RuntimeError("Length of MIBlock5bits:{0}, MIWaveGroup2Bits : {1},MIWaveGroup2Bits:{2} cannot be empty".format(len(MIBlock5bits),len(MIWaveTile2Bits),len(MIWaveGroup2Bits)))
+    
+   MIInstruction9Bits = "         - MatrixInstruction:\n           - ["+ MIBlock5bits + "," + MIWaveTile2Bits + "," + MIWaveGroup2Bits + "]\n"
+   return MIInstruction9Bits
+    
+def FormProblemSizeYamlData(exactLogic,solutionIndex,scheduleName,deviceNames,architectureName):
+    
+    YamlFileData.append("      BenchmarkJoinParameters:\n")
+    YamlFileData.append("      BenchmarkFinalParameters:\n")
+    
+    YamlFileData.append("         - ProblemSizes:\n")
+    #Form the problem Size 
+    for (size, mapping) in exactLogic:
+      if mapping[0] == solutionIndex: 
+        YamlFileData.append("           - Exact: %s             # Eff: %s  Solution Index: %s\n" % (size,mapping[1],mapping[0]))
+        
+     #Form final library logic string
+    YamlFileData.append("#########################################################################################\n")
+    YamlFileData.append("LibraryLogic:\n")
+    YamlFileData.append("    ScheduleName: %s\n"%scheduleName)
+    YamlFileData.append("    DeviceNames: %s\n"%deviceNames)
+    YamlFileData.append("    ArchitectureName: \"%s\"\n"%architectureName)
+        
+def CheckMacroTileThreadTileWorkGroupMatches(currentSolution):
+
+  if not currentSolution.keys() & {'MIBlock','MIWaveTile','MIWaveGroup'}:
+    raise RuntimeError("one or more of these fields => MIBlock, MIWaveTile, MIWaveGroup is missing in the library logic file!! ..\n")

```

Comment:
I guess, if these items are not available, then the config file does not have MatrixInstruction and it is only based on ThreadTile and WorkGroup. For example SolutionIndex: 15, in ../asm_full/arcturus_Cijk_Ailk_Bjlk_SB.yaml. If that is the case, all you need is to add TT and WG. Can you please check this out?

Reply:
yes ,MI* parameters are missing for if the matrix instruction is empty or EnableMatrixInstruction is false
EnableMatrixInstruction: false
MatrixInstruction: []